Appsflyer : Update regarding support for all sources. (#322)
rudderlabs/rudderstack-docs: destinations/attribution/appsflyer.mdx

For more information on this field can be found in the <a href="https://support.appsflyer.com/hc/en-us/articles/207034486-Server-to-server-events-API-for-mobile-S2S-mobile-#sharing_filter-16">AppsFlyer documentation</a>.
</div>

> - **App ID**: RudderStack supports all the sources for the AppsFlyer destination. If your source is other than **Android** or the **Apple** sources (iOS, watchOS, ipados, tvOS) then transformation in dashboard is needed to manipulate the os name.

Reviewer: ```suggestion
- **App ID**: RudderStack supports all the sources for the AppsFlyer destination. If your source is other than **Android** or the **Apple** sources (iOS, watchOS, iPadOS, tvOS) then a user transformation in the dashboard is needed to manipulate the `os` name.
```

> - **App ID**: RudderStack supports all the sources for the AppsFlyer destination. If your source is other than **Android** or the **Apple** sources (iOS, watchOS, iPadOS, tvOS) then a user transformation in the dashboard is needed to manipulate the `os` name and version.

Reviewer: ```suggestion
- **App ID**: RudderStack supports all the sources for the AppsFlyer destination. If your source is other than **Android** or the **Apple** sources (iOS, watchOS, iPadOS, tvOS) then a user transformation in the dashboard is needed to manipulate the OS information.
```

  - **Android App ID**: This the application ID used in your `app/build.gradle` file.
  - **Apple App ID**: This is the iTunes Application ID and it is mandatory for the iOS applications.

<div class="infoBlock">

> RudderStack supports all the other sources for the AppsFlyer destination, in addition to the Android and iOS sources. Refer to the <a href="#connecting-appsflyer-to-non-androidios-sources">Connecting AppsFlyer to non-Android/iOS sources</a> section below for more information.

Reviewer: ```suggestion
RudderStack supports all the other sources for the AppsFlyer destination, in addition to the Android and iOS sources. Refer to the <a href="#connecting-appsflyer-to-non-androidios-sources">Connecting AppsFlyer to non-Android/Apple sources</a> section below for more information.
```

    </TabPanels>
</Tabs>

> ## Connecting AppsFlyer to non-Android/iOS sources

Reviewer: ```suggestion
## Connecting AppsFlyer to non-Android/Apple sources
```

As mentioned above, the AppsFlyer destination supports all the RudderStack sources in addition to Android and iOS. However, you will need to add a transformation to the destination by following these steps:

- Go to your [**RudderStack dashboard**](https://app.rudderstack.com/).

- Add a new transformation by following the instructions listed in [**this guide**](https://rudderstack.com/docs/transformation/#adding-a-transformation). Use the following transformation function:

```javascript
export function transformEvent(event, metadata) {
  event.context.os = {
    name: "android"

>   };

Reviewer: @desusai7 is context.os.version required for AppsFlyer destination? I see that we populate it in the payload

Reviewer 2: Yeah, we need it for correct data processing as mentioned [here](https://support.appsflyer.com/hc/en-us/articles/207034486-Server-to-server-events-API-for-mobile-S2S-mobile-#os-20)
More details [here](https://support.appsflyer.com/hc/en-us/articles/207034486-Server-to-server-events-API-for-mobile-S2S-mobile-#-required-by-many-ad-networks-for-optimization-purposes-to-enable-correct-data-processing-you-must-send-this-parameter-for-backward-compatibility-reasons-we-dont-enforce-this-so-it-isnt-marked-as-obligatory-22)

Reviewer: Thanks, @desusai7. @ameypv-rudder let's update to mention that version is also required.

Author: have updated doc to include version as well.

    </TabPanels>
</Tabs>

## Connecting AppsFlyer to non-Android/iOS sources

As mentioned above, the AppsFlyer destination supports all the RudderStack sources in addition to Android and iOS. However, you will need to add a transformation to the destination by following these steps:

- Go to your [**RudderStack dashboard**](https://app.rudderstack.com/).

- Add a new transformation by following the instructions listed in [**this guide**](https://rudderstack.com/docs/transformation/#adding-a-transformation). Use the following transformation function:

```javascript
export function transformEvent(event, metadata) {
  event.context.os = {
    name: "android",

>     versions: "8.1.0"

Reviewer: ```suggestion
    version: "8.1.0"
```
